app/controllers/application_controller.rb

```diff
@@ -2,4 +2,57 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  #USER AUTH
+
+  def current_user
+     User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def is_authenticated?
+    !!session[:user_id]
+  end
```

Comment:
This should probably be !! current_user. What if there's a value in the session under :user_id but it's not valid?